src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -2435,6 +2458,14 @@ public interface FormDefProcessor extends Processor {
         void processFormDef(FormDef formDef) throws ParseException;
     }
 
+    public interface BindProcessor extends Processor {
+        void processBind(@NotNull DataBinding binding);
+    }
+
+    /**
+     * @deprecated Use {@link BindProcessor} instead
+     */
+    @Deprecated
```

Comment:
We could potentially just remove this, but given someone could theoretically be using it, we'd need to a major version bump.

Reply:
I'm having trouble understanding the `BindAttributeProcessor` -> `BindProcessor` change. Currently `BindAttributeProcessor` makes it possible to create a processor for a specific bind attribute and namespace. It feels like creating a processor for `calculate` in the same way would be fine and good. I'm not seeing any advantage to the `BindProcessor` approach.

Reply:
I'd written a comment in the PR description about this:

> We could have used the existing BindAttributeProcessor, but that would have meant parsing calculate in the processor which didn't feel ideal.

Does that make sense? Happy to discuss further. I actually couldn't remember why I made this choice until I read that!

Reply:
Is this saying that the `value` for the `calculate` attribute is raw so it would require parsing? I think it would be ok for a `BindAttributeProcessor` to use the `DataBinding` that it's given to access the parsed `calculate` and ignore the `value`. Maybe `BindAttributeProcessor` could get a `processBindAttribute(String name, DataBinding binding)` method that's preferred for attributes that are already parsed into the `DataBinding`?

Reply:
> Is this saying that the value for the calculate attribute is raw so it would require parsing?

Yeah exactly. The `BindAttributeProcessor` would have to parse the contents of `calculate` (which would just be a string).

> I think it would be ok for a BindAttributeProcessor to use the DataBinding that it's given to access the parsed calculate and ignore the value. Maybe BindAttributeProcessor could get a processBindAttribute(String name, DataBinding binding) method that's preferred for attributes that are already parsed into the DataBinding?

Ah I guess it could! I think I still prefer having a `BindProcessor` as it just feels simpler to me. You hook into every bind and can make your decision about what you want to do rather than having to implement (and understand) `getBindAttributes` and `processBindAttribute`. Are you just feeling that the change isn't worth the disruption?

Reply:
I'll investigate implementing this with `BindAttributeProcessor`.